packages/tree-select/src/index.ts

```diff
@@ -76,7 +110,8 @@ const NULLISH_KEY = {};
  * Note: Inserts WeakMaps except for the last map which will be a regular Map.
  * The last map is a regular one because the the key for the last map is the string results of args.join().
  */
-function insertDependentKey( map, key, currentIndex, arr ) {
+function insertDependentKey( map: any, key: unknown, currentIndex: number, arr: unknown[] ) {
```

Comment:
This is an unusual reducer function, it consumes an array of keys to walk a tree data structure (`WeakMaps` of `WeakMaps`) until it reaches a leaf which is a `Map< string, CachedSelectorResultValue >`.

I've typed `map` as `any` because usage suggests this internal function is working well enough and it was difficult to type it as `Map<…> | WeakMap<…>` here.